sniffio/_impl.py

```diff
@@ -1,12 +1,27 @@
 from contextvars import ContextVar
-from typing import Optional
+from typing import Optional, TYPE_CHECKING
 import sys
 
+if TYPE_CHECKING or sys.version_info >= (3, 7):
+    import threading
+else:
+    try:
+        import threading
+    except ImportError:  # pragma: no cover
+        import dummy_threading as threading  # pragma: no cover
+
 current_async_library_cvar = ContextVar(
     "current_async_library_cvar", default=None
 )  # type: ContextVar[Optional[str]]
 
 
+class _CurrentAsyncLibraryTLocal(threading.local):
+    name = None  # type: Optional[str]
```

Comment:
So this is creating a class attribute as a default?  Hmm, I guess since all other assignments will be thread-specific that kind of makes sense.  Even if it's not something I normally like.  Assuming I got that right, might be worth a comment?

```suggestion
    # Since threading.local provides no explicit mechanism is for setting
    # a default for a value, a custom class with a class attribute is used
    # instead.
    name = None  # type: Optional[str]
```

Reply:
yeah it's here for two reasons - one for the 3.5 style annotation and one to set the default

```python
class _CurrentAsyncLibraryTLocal(threading.local):
    def __init__(self):
        self.name = None  # type: Optional[str]
```

would work just as well, I think.

Reply:
Yeah, the `class Whatever(threading.local): blah = None` thing is weird, but it's the standard idiom when dealing with `threading.local`.